src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -311,6 +312,26 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		return decoration;
 	}
 
+	registerMenuItems(command: ITerminalCommand, items: IAction[]): IDisposable {
+		const existingItems = this._registeredMenuItems.get(command);
```

Comment:
It looks like the check for existingItems doesn't account for the case when it's an empty array. Since an empty array is considered truthy, it would be better to verify if existingItems exists and has elements by checking existingItems.length. This way, we can avoid unintended behavior when pushing items into an empty array.